flink-ml-core/src/test/java/org/apache/flink/ml/util/ReadWriteUtilsTest.java

```diff
@@ -0,0 +1,107 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.util;
+
+import org.apache.flink.api.common.restartstrategy.RestartStrategies;
+import org.apache.flink.ml.api.ExampleStages;
+import org.apache.flink.ml.api.Model;
+import org.apache.flink.ml.api.Stage;
+import org.apache.flink.ml.api.StageTest;
+import org.apache.flink.ml.api.TestUtils;
+import org.apache.flink.ml.builder.PipelineModel;
+import org.apache.flink.streaming.api.environment.ExecutionCheckpointingOptions;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.test.util.AbstractTestBase;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** Tests {@link ReadWriteUtils}. */
+public class ReadWriteUtilsTest extends AbstractTestBase {
+
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+    private MiniDFSCluster hdfsCluster;
+
+    @Before
+    public void before() throws IOException {
+
+        org.apache.flink.configuration.Configuration config =
+                new org.apache.flink.configuration.Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        Configuration conf = new Configuration();
+        conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, tempFolder.newFolder().getAbsolutePath());
+        MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(conf);
+        hdfsCluster = builder.build();
+    }
+
+    @After
+    public void after() {
+        hdfsCluster.shutdown();
+    }
+
+    @Test
+    public void testModelSaveLoad() throws Exception {
+        // Builds a PipelineModel that increments input value by 60. This PipelineModel consists of
+        // three stages where each stage increments input value by 10, 20, and 30 respectively.
+        ExampleStages.SumModel modelA =
```

Comment:
Thanks for the update.

Would it be simpler to just save and load one SumModel instance, instead of saving/loading a pipeline of three SumModel instances here?

Reply:
Hmm... even if there is only one model, we can still set its model data, save it and load it, right? Do you mean there is something we can not cover by just saving/loading one `SumModel` instance?

Reply:
Let me revise it